scripts/createKafkaTopics.yaml

```diff
@@ -37,18 +37,31 @@ spec:
 
               KAFKA_CONFIG_FILE="/tmp/kafka-endpoint.config"
               cat <<EOT >> ${KAFKA_CONFIG_FILE}
-              sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="token" password="${KAFKA_APIKEY}";
               security.protocol=SASL_SSL
-              sasl.mechanism=PLAIN
               ssl.protocol=TLSv1.2
               ssl.enabled.protocols=TLSv1.2
               ssl.endpoint.identification.algorithm=HTTPS
               EOT
+              
+              if [[ "${KAFKA_USER}" == "token" ]]; then 
+                echo "sasl.mechanism=PLAIN" >> ${KAFKA_CONFIG_FILE}
+                echo "sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username=\"${KAFKA_USER}\" password=\"${KAFKA_PASSWORD}\";" >> ${KAFKA_CONFIG_FILE}
+              else
+                echo "ssl.truststore.type=PKCS12" >> ${KAFKA_CONFIG_FILE}
+                echo "sasl.mechanism=SCRAM-SHA-512" >> ${KAFKA_CONFIG_FILE}
+                echo "sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username\=\"${KAFKA_USER}\" password\=\"${KAFKA_PASSWORD}\";" >> ${KAFKA_CONFIG_FILE}
+              fi
 
               if [[ -f "${TRUSTSTORE_PATH}" ]]; then
                 echo "ssl.truststore.location=${TRUSTSTORE_PATH}" >> ${KAFKA_CONFIG_FILE}
                 echo "ssl.truststore.password=${TRUSTSTORE_PWD}" >> ${KAFKA_CONFIG_FILE}
               fi
+              
+              echo "------------------------------------------"
+              echo "-- Connection Properties --"
+              echo "------------------------------------------"
+              cat ${KAFKA_CONFIG_FILE}
```

Comment:
This should be removed since the logs will get streamed and contain the Kafka passwords

Reply:
I commented it out so that credentials do not get printed in logs but left the code in there as it could be handy for debugging purposes.